contracts/Collector.sol

```diff
@@ -9,45 +9,47 @@ import "@openzeppelin/contracts/math/SafeMath.sol";
 contract Collector is ICollector {
     address private _remainderAddress;
     RevenuePartner[] private _partners;
-    IERC20 public token;
+    IERC20[] private _tokens;
     address public owner;
 
-    modifier validShares(RevenuePartner[] memory partners) {
-        uint256 totalShares;
-
-        for (uint256 i = 0; i < partners.length; i++) {
-            require(partners[i].share > 0, "0 is not a valid share");
-            totalShares = totalShares + partners[i].share;
-        }
-
-        require(totalShares == 100, "Shares must add up to 100%");
-        _;
-    }
-
     modifier onlyOwner() {
         require(msg.sender == owner, "Only owner can call this");
         _;
     }
 
     modifier noBalanceToShare() {
```

Comment:
This only adds the percentages, not the actual values.

Reply:
Sorry Github noob here. I meant the `noBalanceToShare` modifier. If I am not mistaken, following `require` would fail if there is any token left to share. Attacker can send small amount of tokens to the Collector contract to make this condition fail deliberately. Of course, the risk is.... questionable. 

```solidity
    require(
                _tokens[i].balanceOf(address(this)) < _partners.length,
                "There is balance to share"
            );
```

Reply:
@ulasanil thank you for pointing it out. We do agree that there's a risk here, but due to its level, we'll mitigate it in a separate task.